### DISCUSSION TITLE
Script: OBIS decoder with change of baud rate and identifier

### DISCUSSION THREAD

**Discussion by JaFeLi123** [arendst/Tasmota#20433] [Q&A]
Hi everyone,
I hope this is the correct place to ask this question and sorry if this has been answered before. I have not found it.

I want to read a heat meter (Landis&Gyy T550) using the Tasmota script interface and Smart Meter Language through its optical interface that is somehow like EN 62056-21. There is a good (german) description at http://www.sedelmaier.at/node/112 with a Python script as demonstrator. In short:

- Serial Mode is 7E1
- You have to send 40 0x00 bytes with baud 300 to wake up the meter
- You have to request data with "\x2F\x3F\x21\x0D\x0A" and wait some time to read the identifier (b'/LUGCUH50\r\n')
- After this you have to switch to baud 2400 and you'll get the OBIS data.

The python script outputs the following (first line is read with baud 300):
```
b'/LUGCUH50\r\n'
b'\x026.8(0003522*kWh)6.26(00098.06*m3)9.21(71740034)\r\n'
b'6.26*01(00090.24*m3)6.8*01(0003190*kWh)\r\n'
b'F(0)9.20(71740034)6.35(60*m)\r\n'
....
b'!\r\n'
```
I have converted this into the following script that at least gives me a response form the meter:
```
>D

>B
;setup sensor
->sensor53 r

>S
if upsecs%15==0 {
print read meter
=#readmeter
}
#

#readmeter

;set serial protocol
sml(1 0 300)

;send wakeup sequence
sml(1 1 "0000000000000000000000000000000000000000")
sml(1 1 "0000000000000000000000000000000000000000")
sml(1 1 "2F3F210D0A")
delay(200)
print wakeup end

;switch serial protocol
sml(1 0 2400)


>M 1
+1,44,o,0,2400,WAERME,43
1,6.26(@1,Test,m3,v_test,2
1,6.8(@1,Tot. Energy,kWh,w_total,0
#
```
The problem is that the first two lines of the python script are not succesfully received by Tasmota. I do get the following output in the console:
```
22:41:06.567 read meter
22:41:08.102 wakeup end
22:41:09.935 : m
22:41:10.134 : 6.26*01(00090.24*m3)6.8*01(0003190*kWh)
22:41:10.278 : F(0)9.20(71740034)6.35(60*m)
.....
22:41:14.852 : !
```
I guess this is somehow related to the identifier and/or the 0x02 in front of the first OBIS line?

Help is very much appreciated and I would also update the Tasmota Doc once this is running.

Best,
Jan

Edit: Sorry I tried but I do not get the Code-Preview with Markdown working...

**Comment by sfromis** [comment]
Code blocks with markdown is triple-backtick lines before and after.
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks
Location of that key depends on your language/keyboard layout. Assuming German, it would likely be shift and the top right key before the backspace key, and a space to get it out of being a dead key.

**Reply by JaFeLi123** [reply]
Thanks a lot. The editor was just inserting single backticks.
Triple-backticks are working and initial post is edited.
